src/components/Image/index.tsx

```diff
@@ -0,0 +1,86 @@
+import styled from '@emotion/styled'
+import { useRef, useEffect, useState } from 'react'
+
+interface IImage {
+  src: string
+  block: boolean
+  iscircle: boolean
+  style?: React.CSSProperties
+  mode?: 'fill' | 'contain' | 'cover'
+  width?: number
+  height?: number
+  lazy?: boolean
+  threshold?: number
+  placeholder?: string
+}
+
+let observer = null
+const LOAD_IMG_EVENT_NAME = 'loadImage'
+
+const onIntersection: IntersectionObserverCallback = (entries, io) => {
+  entries.forEach((entry: IntersectionObserverEntry) => {
+    if (entry.isIntersecting) {
+      io.unobserve(entry.target)
+      entry.target.dispatchEvent(new CustomEvent(LOAD_IMG_EVENT_NAME))
+    }
+  })
+}
+
+const Image = ({
+  src,
+  block,
+  iscircle,
```

Comment:
block과 isCircle을 옵션 props로 지정하고 default 값을 주면 default 상황에서는 props 전달을 줄일 수 있지 않을까요?

Reply:
```suggestion
  display = 'block',
  iscircle = false,
```
display 인터페이스를 string으로 받아서 block, inline, inline-block, flex, grid ... 등등 가능하도록 인터페이스 변경하였습니다